src/Database/Driver/Postgres.php

```diff
@@ -291,6 +293,20 @@ protected function _transformFunctionExpression(FunctionExpression $expression):
         }
     }
 
+    /**
+     * Changes string expression into postgresql format.
+     *
+     * @param \Cake\Database\Expression\StringExpression $expression The string expression to tranform.
+     * @return void
+     */
+    protected function _transformStringExpression(StringExpression $expression): void
+    {
+        $collation = $expression->getCollation();
+        if ($collation) {
+            $expression->setCollation('"' . trim($collation, '"') . '"');
```

Comment:
The `trim()` works around a design flaw where expressions are transformed every time `sql()` is called. I don't know how to fix that in 4.2.